### PR TITLE
Feature/tests-environment-setup

### PR DESCRIPTION
**Pull request by javierriveros**
Initial configuration for the test environment.
The test environment is working now.
Github Actions is working.

### CODE REVIEW

File: testapp/settings.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+if os.environ.get('GITHUB_WORKFLOW'):
+    DATABASES = {
+        'default': {
+           'ENGINE': 'django.db.backends.postgresql',
+           'NAME': 'djangoplicity-customsearch',
+           'USER': 'djangoplicity-customsearch',
+           'PASSWORD': 'djangoplicity-customsearch',
+           'HOST': '127.0.0.1',
+           'PORT': '5432',
+        }
```
Comment (EdisonArango) [inline testapp/settings.py]: Great that you have the setup with postgres already 👍

File: .github/workflows/django.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest-django
+      - name: Run migrations
+        run: python manage.py migrate
+      - name: Run tests
+        run: |
+          coverage run --source='.' manage.py test
+          coverage xml
```
Comment (EdisonArango) [inline .github/workflows/django.yml:49]: If you are only using here Django tests `(manage.py test)`, then why would you need pytest?.

File: tests/runtests.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
```
Comment (EdisonArango) [inline tests/runtests.py]: If you believe running the tests using this file with the django.setup(), then using `./manage.py test` is not required and you can instead use this file, but if you prefer the django project way using `./manage.py test` then this file is not required.

File: testapp/settings.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
+"""
+Django settings for testapp project.
```
Comment (EdisonArango) [inline testapp/settings.py]: This is not a `testapp`, this is a Django project, a Django app is a package with models, views, tests, etc.

File: requirements/requirements-testing.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+pytest>=4.6.1,<4.7 # 4.6 is the last version to support python 2.7
+pytest-django==3.9.0
+pytest-cov==2.9.0 # coverage report
```
Comment (EdisonArango) [inline requirements/requirements-testing.txt]: If you won't use pytest but only django tests then these dependencies are not required.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  web:
+    build: .
+    container_name: djangoplicity-customsearch-web
+    command: ["./scripts/command-dev.sh"]
+    volumes:
+      - ./djangoplicity:/app/djangoplicity
+      - ./testapp/:/app/testapp
+      - ./htmlcov:/app/htmlcov
+      - ./.coveragerc:/app/.coveragerc
+      - ./db:/app/db
```
Comment (EdisonArango) [inline docker-compose.yml]: This db volume was intended to save a sqlite database, but as you are using postgres then it's not required, you should rather create a named volume for the postgres database.

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,2 @@
+testCoverage:
+	docker exec -it djangoplicity-customsearch-web pytest tests --cov=djangoplicity
```
Comment (EdisonArango) [inline Makefile]: Again, please keep homogeneity by running the tests the same way in all the parts, then use here the django tests instead: `./manage.py test`

File: testapp/settings.py (hunk 4 above)
Comment (javierriveros) [reply testapp/settings.py]: So, what should be the name?, This was generated by django when I created the app.

Comment (EdisonArango) [reply testapp/settings.py]: Again, it's not an `app`, it's a project, then better to call it: `testproject` or `test_project`

**Pull request merged**